libgadget/fof.c

```diff
@@ -232,35 +232,46 @@ struct FOFPrimaryPriv {
 #define FOF_PRIMARY_GET_PRIV(tw) ((struct FOFPrimaryPriv *) (tw->priv))
 
 static int
-HEADl(int stop, int i, TreeWalk * tw)
+HEADl(int stop, int i, int locked, int * Head)
```

Comment:
Now that it returns three possible values probably write a comment -1 is found the stop and -2 is lock failed.

Reply:
Good idea, will add it.